Xabe.FFmpeg/Conversion/Conversion.cs

```diff
@@ -342,6 +344,24 @@ private string BuildInput()
             return builder.ToString();
         }
 
+        /// <summary>
+        /// Builds the output path with an optional number specifier
+        /// </summary>
+        /// <param name="digits"> the number of digits to include in the file number</param>
+        /// <param name="position"> the position of the file number within the output path </param>
+        /// <returns></returns>
+        public string BuildNumberedOutput(NumberPosition position = NumberPosition.BEGIN, int digits = 3)

```

Comment:
It won't work. 
At the moment when you try to set up this BuildNumberedOutput, the _output variables is empty.
You have to declare delegate which will be used to build output file name:

        private Func<string, string> buildOutputFileName = (number =>
                                                            {
                                                                return "File_No{number}.png";
                                                            }); 

Something like this

Reply:
And User will define this function in argument in ExtractNthFrame/ExtractEveryNthFrame parameter

Reply:
OK I think I understand now. So the result of this delegate would be passed into SetOutput function within the ExtractNthFrame/ExtractEveryNthFrame functions?

Reply:
It should be passed into ExtractNthFrame/ExtractEveryNthFrame.
Next instead of builder.Append(_output); in line 49, it should use output of this function. Default it will return _output, but if user use the Extract function it will be overrided

Reply:
Ok thanks makes sense now. One last thing how does this handle multiple outputs without a %3d?

Reply:
This function will looks like this:

private Func<string, string> buildOutputFileName = (number =>
                                                        {
                                                            return "File_No{number}.png";
                                                        }); 

User will have to use somewhere this number